src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1377,6 +1380,8 @@ private function dumpValue($value, $interpolate = true)
                 // we do this to deal with non string values (Boolean, integer, ...)
                 // the preg_replace_callback converts them to strings
                 return $this->dumpParameter(strtolower($match[1]));
+            } elseif (preg_match('/^\$([^\$]+)\$$/', $value, $match)) {
```

Comment:
This is will also match `$foo'boom$` making it possible to unwillingly inject PHP code.

And the $$ will not match but is not normalized to a single $ in the else-block.